lib/active_hash/base.rb

```diff
@@ -157,14 +157,29 @@ def all(options={})
       def where(options)
         return @records if options.nil?
         (@records || []).select do |record|
-          options.all? { |col, match| record[col] == match }
+          match_options?(record, options)
         end
       end
 
       def find_by(options)
-        where(options).first
+        return all.first if options.nil?
```

Comment:
When calling `find_by` with no arguments, this will act like `first`, which seems strange.  Previously it would return `nil`.  Can you just return `nil` instead of returning `all.first`?